drgn_tools/crash_net.py

```diff
@@ -0,0 +1,372 @@
+        for dev_node in hlist_for_each_entry(entry_type, head, member):
+            dev = dev_node.dev if entry_is_name_node else dev_node
+            yield (dev_node.name.string_().decode("utf-8"), dev)
+
+
+def net_crash(prog: Program) -> None:
+    """
+    Display the network devices, their names and addresses.
+    :param prog: drgn program
+    :returns: None.
+    """
+    rows = [["NET_DEVICE", "NAME", "IP ADDRESS(ES)"]]
+
+    net_namespaces = network.for_each_net(prog)
+    for net_namespace in net_namespaces:
+        for name, dev in for_each_netdev(net_namespace):
+            net_device_ptr = hex(dev.dev.platform_data)
+            ip_list = [str(a) for a in netdev_ipv4s(dev) + netdev_ipv6s(dev)]
+            ips = ", ".join(ip_list)
+            rows.append(
+                [
+                    net_device_ptr,
+                    name,
+                    ips,
+                ]
+            )
+    print_table(rows)
+
+
+def crash_a(prog: Program) -> None:
+    """
+    Displays the arp cache.
+    :param prog: drgn program
+    :returns: None.
+    """
+
+    rows = [
+        ["NEIGHBOUR", "IP ADDRESS", "HW TYPE", "HW ADDRESS", "DEVICE", "STATE"]
+    ]
+
+    nht = prog["arp_tbl"].nht
+
+    hash_shift = nht.hash_shift.value_()
+    nhash_buckets = 1 << hash_shift
+    hash_buckets = nht.hash_buckets
+
+    for i in range(nhash_buckets):
+        neigh = hash_buckets[i]
+        while neigh:
+            neigh_addr = hex(neigh.value_())[2:]
+            ip_addr = ""
+            for j in range(4):
+                ip_addr += str(neigh.primary_key[j].value_()) + "."
+            ip_addr = ip_addr[:-1]
+
+            hw = neigh.ha
+            hw_addr = ""
+            for j in range(6):
+                hw_addr += str(hex(hw[j])[2:].zfill(2)) + ":"
+            hw_addr = hw_addr[:-1]
+            dev_name = neigh.dev.name.string_().decode()
+
+            hw_type = str(hwtype_dict.get(int(neigh.dev.type)))
+            if int(neigh.dev.type) not in hwtype_dict.keys():
+                hw_type = "ARPHRD_UNKNOWN"
+            hw_type = hw_type[7:]
+            dev_state = str(state_dict.get(int(neigh.nud_state.value_())))[4:]
+            rows.append(
+                [
+                    neigh_addr,
+                    ip_addr,
+                    hw_type,
+                    hw_addr,
+                    dev_name,
+                    dev_state,
+                ]
+            )
+            neigh = neigh.next
+    print_table(rows)
+
+
+def print_task_sockets(prog: Program, pid: int, print_full_data: bool) -> None:
+    """
+    Displays the socket addreses, family type, source and destination
+    addresses and corresponding ports along with the dump of data
+    structures. If print_full_data is true, it prints detailed data
+    structure dumps.
+    :param prog: drgn program
+    :param pid: pid of the process.
+    :returns: None.
+    """
+
+    task = find_task(prog, int(pid))
+    if not (task):
```

Comment:
nit: no need for parantheses - `if not task:` is the standard way of expressing this